woocommerce/woocommerce
# Rating Star: render a single SVG wrapper element per stars group

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR changes the markup generated by the `<StarRating />` react component, returning a single (wrapper) SVG element, instead of one SVG per star. It could be considered a polishing PR since it doesn't change the component's functionality.

Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Ensure the experimental blocks are enabled:
    - If you use the WooCommerce Beta Tester plugin, you can enable the experimental blocks in Tools > WCA Test Helper > Features. Toggle the `experimental-blocks` option.
    - If you build this PR locally, build the branch with `pnpm --filter='@woocommerce/plugin-woocommerce' build` command to have experimental blocks available.
1. Ensure to add a few products to your store
2. Ensure to rating products
3. Add the **Product Filters (Experimental)** block to the Product Catalog template.
4. Select the `Rating (Experimental)` block

<img width="400" alt="image" src="https://github.com/user-attachments/assets/d0e01490-3e86-4cbc-9737-925b0e2b8614">

5. Open the block settings (sidebar)
6. Confirm you see the `Minimum rating` panel

<img width="289" alt="image" src="https://github.com/user-attachments/assets/0548af90-8ce7-432e-9b4c-2667bfdc084b">

8. Confirm you can select different options from there
9. Save the template
10. Hard refresh
11. Confirm the changes are persistent

You can take a look at the elements generated by the component, confirming that before it renders a `<SVG />` per star.
After these changes, it generates only one `<SVG />` wrapper element

before | after
-----|-----
<img width="708" alt="Screenshot 2024-11-18 at 12 43 48 PM" src="https://github.com/user-attachments/assets/4ebeafa8-1997-4043-a485-aa6e96c6af38"> | <img width="581" alt="Screenshot 2024-11-18 at 12 50 43 PM" src="https://github.com/user-attachments/assets/f21ca15f-97c6-4e76-8140-e421e2896420">

## With storybook

1. Go to the woocommerce blocks folder

```
cd plugins/woocommerce-blocks
```

2. Start up the Storybook app

```
pnpm run storybook
```

3. Play with the RatingStars story
4. Confirm it works as expected:


https://github.com/user-attachments/assets/ebc8ad4a-6444-4742-b2b7-6e358688bd33





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
